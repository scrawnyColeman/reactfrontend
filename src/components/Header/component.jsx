import React from 'react';
import styled from 'styled-components';
import { colours } from '../../constants/styles';
import Logo from '../Logo/component';

const StyledHeader = styled.div`
    position: fixed;
    top: 0;
    left: 0;
    padding: 12px 0;
    width: 100%;
    display: grid;
    grid-template-columns: 2fr 1fr;
    grid-gap: 5px;
    text-transform: uppercase;
    @media only screen and (max-width: 968px) {
        display: block;
    }
`;
export const StyledNavLinks = styled.a`
    color: ${colours.primary};
    padding: 10px 0;
    font-size: 1.25rem;
    text-decoration: none;
`;
const StyledNavBar = styled.div`
    display: flex;
    justify-content: space-around;
`;
const StyledLogoContainer = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    @media only screen and (max-width: 968px) {
        display: none;
    }
`;

const Header = ({ userName }) => (
    <StyledHeader>
        <StyledNavBar>
            <StyledNavLinks href="/home">HOME</StyledNavLinks>
            <StyledNavLinks href="/learn">LEARN</StyledNavLinks>
            <StyledNavLinks href="/forum">FORUM</StyledNavLinks>
            {userName ? (
                <StyledNavLinks href="/login">{userName}</StyledNavLinks>
            ) : (
                <StyledNavLinks href="/login">Log in</StyledNavLinks>
            )}
        </StyledNavBar>
        <StyledLogoContainer>
            <Logo />
        </StyledLogoContainer>
    </StyledHeader>
);

export default Header;
