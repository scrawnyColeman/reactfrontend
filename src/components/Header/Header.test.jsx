import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import Header, {StyledNavLinks} from './component';

Enzyme.configure({ adapter: new Adapter() });

describe('Header component', () => {
    global.window = {
        location: { href:"/" },
    };
    test('component matches snapshot', () => {
        const wrapper = shallow(<Header userName="Thomas"/>);
        expect(wrapper).toMatchSnapshot();
    });

    test.each`
        linkNumber | expectedHref
            ${0} | ${'/learn'}
            ${1} | ${'/forum'}
            ${2} | ${'/profile/{id}'}
    `('href is set onClick of navigation link', ({ linkNumber, expectedHref }) => {
        const wrapper = shallow(<Header userName="Thomas"/>);
        expect(wrapper.find(StyledNavLinks).at(linkNumber).props().href).toBe(expectedHref);
    });
});
