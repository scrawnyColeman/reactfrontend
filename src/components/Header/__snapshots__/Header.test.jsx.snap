// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component component matches snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.a
      href="/home"
    >
      HOME
    </styled.a>
    <styled.a
      href="/learn"
    >
      LEARN
    </styled.a>
    <styled.a
      href="/forum"
    >
      FORUM
    </styled.a>
    <styled.a
      href="/login"
    >
      Thomas
    </styled.a>
  </styled.div>
  <styled.div>
    <Logo />
  </styled.div>
</styled.div>
`;
